src/services/depositService.ts

```diff
@@ -271,6 +273,45 @@ export async function getMinimumStorageBalance(
   })
 }
 
+export async function getAllowance(
+  tokenAddress: string,
+  owner: string,
+  spender: string,
+  network: BlockchainEnum
+): Promise<bigint | null> {
+  try {
+    const client = createPublicClient({
```

Comment:
I don't like creating an rpc client in place, since it may send unnecessary request every time it is created. For example, ethers provider sends eth_chainId request for the first call. It's place to think of and generalize the approach so it can be used in other place as well.